api/iteration_test.go

```diff
@@ -1,8 +1,57 @@
 package api
 
-import "testing"
+import (
+	"path/filepath"
+	"runtime"
+	"testing"
+)
 
-func TestIdentify(t *testing.T) {
+func TestNewIteration(t *testing.T) {
+	_, path, _, _ := runtime.Caller(0)
+	dir := filepath.Join(path, "..", "..", "fixtures", "iteration")
+
+	files := []string{
+		filepath.Join(dir, "python", "leap", "one.py"),
+		filepath.Join(dir, "python", "leap", "two.py"),
+		filepath.Join(dir, "python", "leap", "lib", "three.py"),
+	}
+
+	iter, err := NewIteration(dir, files)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if iter.Language != "python" {
+		t.Errorf("Expected language to be python, was %s", iter.Language)
+	}
+	if iter.Problem != "leap" {
+		t.Errorf("Expected problem to be leap, was %s", iter.Problem)
+	}
+
+	if len(iter.Solution) != 3 {
+		t.Fatalf("Expected solution to have 3 files, had %d", len(iter.Solution))
+	}
+
+	expected := map[string]string{
+		"one.py":       "# one\n",
+		"two.py":       "# two\n",
+		"lib/three.py": "# three\n",
+	}
+	for filename, code := range expected {
+		if iter.Solution[filename] != code {
+			t.Errorf("Expected %s to contain %s, had %s", filename, code, iter.Solution[filename])
+		}
+	}
+}
+
+/*
```

Comment:
Oh, ouch, that was a left-over something-or-other. Let me figure out what happened and get rid of it.